File: front/src/app/services/reserva.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';

import { Reserva } from '../models/reserva';

@Injectable({
  providedIn: 'root'
})
export class ReservaService {

  constructor(private httpClient: HttpClient) { }

  findAll(): Observable<Reserva[]> {
    return this.httpClient.get<Reserva[]>(`${environment.API}/reservas`)
  }

  findById(id: any): Observable<Reserva> {
    return this.httpClient.get<Reserva>(`${environment.API}/reservas/${id}`)
  }

  save(data: Reserva): Observable<Reserva> {
    return this.httpClient.post<Reserva>(`${environment.API}/reservas`, data)
  }

  update(id: any, data: Reserva): Observable<Reserva> {
    return this.httpClient.put<Reserva>(`${environment.API}/reservas/${id}`, data);
  }
}

File: back/src/main/java/com/api/reserva/services/ReservaService.java
package com.api.reserva.services;

import java.util.List;

import com.api.reserva.data.entity.ReservaEntity;
import com.api.reserva.services.exception.ServiceException;
import com.api.reserva.web.rest.request.ReservaRequest;

public interface ReservaService {

	ReservaEntity findById(Long id) throws ServiceException;

	List<ReservaEntity> findAll();

	ReservaEntity save(ReservaRequest entity) throws ServiceException;

	ReservaEntity update(Long id, ReservaRequest entity) throws ServiceException;
}
